src/noit_check_lmdb.c

```diff
@@ -67,11 +73,17 @@ static int noit_check_lmdb_add_attribute(xmlNodePtr root, xmlNodePtr attr, noit_
 static int noit_check_lmdb_add_config(xmlNodePtr root, xmlNodePtr config, noit_lmdb_check_data_t *key_data, MDB_val mdb_data) {
   mtevAssert(root != NULL);
   mtevAssert(config != NULL);
-  if ((!mdb_data.mv_data) || (mdb_data. mv_size == 0)) {
+  char *val = NULL;
+  if ((mdb_data.mv_data == NULL) && (mdb_data.mv_size > 0)) {
+    /* If data is null, but we have a size, something is wrong - just skip
+     * this key */
+    mtevL(mtev_error, "noit_check_lmdb_add_config: got null data for key with a size, skipping\n");
     return 0;
   }
-  char *val = (char *)calloc(1, mdb_data.mv_size + 1);
-  memcpy(val, mdb_data.mv_data, mdb_data.mv_size);
+  if ((mdb_data.mv_data != NULL) && (mdb_data.mv_size > 0)) {
```

Comment:
no else here what happens if it is null and zero or if it is non-null but zero size but something is there - can that happen?  Note that above you had one case when it was NULL where you malloc'ed a single byte into value and and put a zero into it...

Reply:
There's no else because val is already set to NULL, which is what we want. This case is covered in the test suite.

Reply:
ok